internal/statemachine/classic_states.go

```diff
@@ -400,12 +400,18 @@ func (stateMachine *StateMachine) buildGadgetTree() error {
 		sourceURL, _ := url.Parse(classicStateMachine.ImageDef.Gadget.GadgetURL)
 
 		// copy the source tree to the workdir
-		err := osutilCopySpecialFile(sourceURL.Path, gadgetDir)
+		files, err := ioutilReadDir(sourceURL.Path)
```

Comment:
We might want to consider switching to `os.ReadDir()` at some point, as the ioutil one is deprecated since 1.16.